crates/lsp/src/handlers_format.rs

```diff
@@ -244,132 +244,125 @@ fn find_expression_lists(node: &RSyntaxNode, offset: TextSize, end: bool) -> Vec
 
 #[cfg(test)]
 mod tests {
-    use crate::{documents::Document, test::with_client, test::TestClientExt};
+    use crate::{documents::Document, test::new_test_client, test::TestClientExt};
 
-    #[test]
-    fn test_format() {
-        with_client(|client| async {
-            let mut client = client.lock().await;
+    #[tokio::test]
+    async fn test_format() {
+        let mut client = new_test_client().await;
 
```

Comment:
The diff here is painful, but the simplification is quite nice:
- Can now create a `new_test_client()` in every test
- Test itself gets marked as a tokio test (until we can remove tokio)
- `with_client()` is removed, flattening a level of nesting

This should transfer nicely to lsp-server as well, we just remove the awaits, async, etc.